### DISCUSSION TITLE
End to end encryption

### DISCUSSION THREAD

**Discussion by sudoscientific** [toeverything/AFFiNE#5296] [Q&A] [answered]
Hi I've been using Affine a bit and it's amazing. Was just wondering with Affine cloud is there any e2e encryption?

It's difficult to trust an app as fully private if you have access to everything stored on the cloud.

**Comment by sudoscientific** [comment, marked as answer]
Sorry I saw this post on the FAQ that answers my question.

Q: Does AFFiNE provide support for end-to-end encryption?

A: Certainly, AFFiNE extends robust support for end-to-end encryption (E2EE). Our platform prioritizes data security while fostering seamless teamwork within local networks. With AFFiNE, you can confidently leverage E2EE alongside local area network (LAN) collaboration. Moreover, our collaboration framework, based on Conflict-free Replicated Data Types (CRDTs), ensures that real-time collaboration remains unhindered by geographical constraints, network speeds, and even offline usage. We anticipate these improvements to be rolled out by the end of this year.

**Comment by AbsoluteZero-ljz** [comment]
May I ask which version supports e2ee?

**Reply by remicorniere** [reply]
I'm not sure it does yet. From their FAQ:

> Q: Does AFFiNE provide support for end-to-end encryption?

> A: AFFiNE's data synchronization is based on conflict-free replication data type (CRDT) technology, which is naturally compatible > with end-to-end encryption (E2EE). CRDT also ensures that real-time collaboration is not hindered by geographical restrictions,  > network speed, and even offline use. AFFiNE gives priority to data security and hopes to promote seamless teamwork within > the local network. We expect support for E2EE to be launched in the near future. Stay tuned for updates.

I find the wording a little weird too ("support" ? Not implement encryption ? CRDTs are "compatible" with E2E encryption ? Will it be a core feature ?)
